addon/services/ajax.js

```diff
@@ -113,9 +113,9 @@ export default Ember.Service.extend({
         );
 
         if (response instanceof AjaxError) {
-          reject(response);
+          Ember.run.join(null, reject, response);
```

Comment:
I suspect it is to allow for batching of XHR, i suppose if users are doing that they are in power-user land and must already do the right thing....